release.py

```diff
@@ -0,0 +1,212 @@
+"""Increments the Orchestra version and pushes a new release to PyPI.
+
+This script can only be called by orchestra core committers, and will result
+in permission errors otherwise.
+"""
+from argparse import ArgumentParser
+from distutils.version import StrictVersion
+import os
+import re
+import shutil
+from subprocess import check_output, CalledProcessError
+from tempfile import mkdtemp
+
+VERSION_RE = re.compile("__version__ = ['\"]([^'\"]+)['\"]")
+
+
+def main(args):
+    # Verify that our git setup is reasonable and print a loud warning.
+    verify_and_warn()
+
+    # Compute the new version number
+    old_version = get_version('orchestra')
+    new_version = increment_version_number(old_version, args.version_part)
+    print('Updating Orchestra version from {} to {}'
+          .format(old_version, new_version))
+
+    # Update orchestra/__init__.py with the new version
+    update_version('orchestra', new_version, fake=args.fake)
+
+    # Commit the update to the repo
+    commit_and_push(fake=args.fake)
+
+    # Create a new tag for the release
+    tag_str = tag_release(new_version, fake=args.fake)
+
+    # Release the new version on pypi
+    pypi_release(tag_str, fake=args.fake)
+
+
+def verify_and_warn():
+    # verify that we're on the master branch
+    branch_cmd = ['git', 'symbolic-ref', '--short', 'HEAD']
+    cur_branch = check_output(branch_cmd).decode().strip().lower()
+    if cur_branch != 'master':
+        print('This script must be run from the master branch. Exiting.')
+        exit()
+
+    # verify that master is up to date
+    rev_cmd = ['git', 'rev-list']
+    unpushed_cmd = rev_cmd + ['origin/master..']
+    unpushed_commits = check_output(unpushed_cmd).decode().strip()
+    if unpushed_commits:
+        print('Your branch has commits not yet on origin/master. Exiting.')
+        exit()
+
+    unpulled_cmd = rev_cmd + ['..origin/master']
+    unpulled_commits = check_output(unpulled_cmd).decode().strip()
+    if unpulled_commits:
+        print('Your branch is behind origin/master. Exiting.')
+        exit()
+
+    # verify that there are no outstanding changes
+    status_cmd = ['git', 'status', '--porcelain']
+    status_output = check_output(status_cmd).decode().strip()
+    if status_output:
+        print('There are outstanding changes to your branch. Exiting.')
+        exit()
+
+    print('WARNING: running this script will increment the Orchestra version. '
```

Comment:
move this warning until after we know what version we're bumping, and before pypi (can be significantly before)